module1/homework/grayscaleImages/main.cpp

```diff
@@ -1,51 +1,119 @@
+        {0,  0,  0,  0,  0,  0,  0,  24, 35, 47, 46, 48, 51, 51, 52, 54,
+         54, 58, 61, 61, 56, 54, 57, 38, 23, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 44, 51, 44, 45, 51, 51, 51, 51,
+         51, 54, 57, 57, 48, 48, 66, 49, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 44, 51, 44, 45, 51, 51, 51, 51,
+         51, 51, 52, 57, 48, 48, 66, 49, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 44, 51, 44, 45, 51, 51, 51, 51,
+         51, 51, 52, 57, 48, 48, 66, 49, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 44, 51, 44, 45, 51, 51, 51, 51,
+         51, 51, 52, 57, 48, 47, 64, 48, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 44, 51, 44, 45, 51, 51, 51, 51,
+         51, 51, 52, 57, 48, 46, 57, 46, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  35, 43, 44, 42, 60, 85, 85, 85, 85,
+         85, 85, 85, 86, 60, 43, 46, 43, 34, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0,  28, 34, 34, 41, 59, 82, 92, 88, 82,
+         82, 82, 83, 85, 60, 40, 34, 34, 27, 0,  0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 45, 51, 89, 77, 51,
+         51, 51, 53, 66, 51, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 45, 51, 89, 77, 51,
+         51, 54, 57, 66, 51, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 45, 51, 76, 67, 51,
+         51, 54, 58, 66, 51, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 45, 51, 51, 51, 51,
+         51, 54, 58, 66, 51, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 45, 51, 51, 47, 43,
+         43, 51, 58, 66, 51, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0,  0,  0,  0,  0,  0,  0, 0, 0, 0, 37, 44, 47, 47, 40, 29,
+         29, 41, 51, 56, 47, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
+        {0, 0,  0,  0,  0,  0,  0, 0, 0, 0, 36, 41, 41, 41, 28, 0,
+         0, 29, 41, 41, 41, 36, 0, 0, 0, 0, 0,  0,  0,  0,  0,  0},
     };
 }
 
 int main() {
     auto ninja = generateNinja();
-    // printMap(ninja);
+    printMap(ninja);
     auto compressed = compressGrayscale(ninja);
+    compressed_bitmap result_comp{};
+
+    for (const auto& row : ninja) {
+        auto it = row.begin();
+        uint8_t numOfUniques =
+            std::count_if(row.begin(), row.end(), [it](const uint8_t number) mutable {
+                std::advance(it, 1);
+                auto result = number != *it;
+                return result;
+            });
+        std::transform(
+            row.begin(), std::next(row.begin(), numOfUniques),
+            std::back_inserter(
+                result_comp),  // dlaczego back inserter nadpisuje pierwszy ostatni elemen
+                               // w nowej petli ? ;/// nie umiem tego poprawic
```

Comment:
back inserter -> push_back. So when you call back_inerter you call push_back for some ranges.

Reply:
I know that, but I i do not know why there is some weird behaviour:
After first iteration i have, for instance: {{0,10}, {32,1}, {42, 1}, {118, 10}, {0, 10}}, but in next iteration {0,10} will be replaced by {0,8} <-- the first elements of new row. This solution become not very friendly to handle ;d

Reply:
If you dont mind, I will leave it, and back to that when I will have some spare time. I have working solution, but now I'd rather focus on SHM :)